pkg/build/registry/buildconfig/circular.go

```diff
@@ -0,0 +1,83 @@
+package buildconfig
+
+import (
+	"errors"
+
+	"github.com/openshift/origin/pkg/build/api"
+)
+
+// deps contains all image repositories mapped to their dependencies
+var deps = make(map[string]map[string]string)
```

Comment:
since this datastructure is in no way persisted, you need some initialization logic that will rebuild it when the server starts.